[PT-6] Add order history

LonelyMidoriya opened the pull request:
Add an "Order History" page
Implement filtering by categories

LonelyMidoriya committed:
[PT-5] Add auth

Add auth with email and google
Add sign up and log in pages
+
Some fixes

LonelyMidoriya committed:
[PT-6] Add order history

Add an "Order History" page
Implement filtering by categories

AliaksandrHuzich commented on food_app/core/lib/consts/consts.dart:
```diff
@@ -1,39 +1,6 @@
-import 'dart:ui';
```
constants

AliaksandrHuzich commented on food_app/core/lib/consts/consts.dart:
```diff
@@ -1,39 +1,6 @@
 const double googleImageSize = 50;
 const String gitHubUrl = 'https://github.com/LonelyMidoriya/food_app';
 const String googleUrl = 'http://pngimg.com/uploads/google/google_PNG19635.png';
+const List<String> typesOfFood = ['all','burger','chicken','fries','sauce'];
```
разделить

AliaksandrHuzich commented on food_app/core/lib/di/data_di.dart:
```diff
@@ -185,6 +182,13 @@ class DataDI {
     appLocator.registerLazySingleton<AuthRepository>(
       () => AuthRepositoryImpl(
         appLocator.get(),
+        appLocator.get(),
```
типы

AliaksandrHuzich commented on food_app/data/lib/mapper/cart_mapper.dart:
```diff
@@ -8,21 +8,27 @@ class CartMapper {
   CartEntity toEntity(CartModel model) {
     final List<CartItemEntity> cartItems = [];
-    for(CartItemModel itemModel in model.cartItems){
+    for (CartItemModel itemModel in model.cartItems) {
```
одним выражением

AliaksandrHuzich commented on food_app/data/lib/mapper/orders_mapper.dart:
```diff
@@ -0,0 +1,31 @@
+  OrdersMapper(this._cartMapper);
+
+  OrdersEntity toEntity(OrdersModel model) {
+    final List<CartEntity> carts = [];
```
сейм

AliaksandrHuzich commented on food_app/data/lib/provider/firestore_provider.dart:
```diff
@@ -22,6 +22,14 @@ class FirestoreProvider {
   }
 
+  Future<void> updateOrderHistory(
+    Map<String, dynamic> orders,
```
именнованные и обязательные

AliaksandrHuzich replied:
лист

AliaksandrHuzich commented on food_app/data/lib/repository/auth_repository_impl.dart:
```diff
@@ -4,61 +4,77 @@ import 'package:domain/domain.dart';
 
 class AuthRepositoryImpl extends AuthRepository {
   final AuthProvider _authProvider;
+  final SharedPreferences prefs;
```
нейминг

AliaksandrHuzich commented on food_app/data/lib/repository/auth_repository_impl.dart:
```diff
@@ -4,61 +4,77 @@ import 'package:domain/domain.dart';
+        'isLoggedIn',
+        false,
+      );
+      await prefs.setString(
```
..

AliaksandrHuzich commented on food_app/data/lib/repository/cart_repository_impl.dart:
```diff
@@ -21,9 +21,14 @@ class CartRepositoryImpl implements CartRepository {
         .then((value) => value.data());
     if (cartJson == null) {
-      return null;
+      return CartModel(
```
именнованный емпту конструктор

AliaksandrHuzich commented on food_app/data/lib/repository/dishes_repository_impl.dart:
```diff
@@ -61,28 +74,50 @@ class DishesRepositoryImpl implements DishesRepository {
+    if (hasInternet) {
+      listEntities = [];
+      await _firestoreProvider.getAllByType('dishes', type).then(
+        (value) {
```
тип

AliaksandrHuzich commented on food_app/data/lib/repository/dishes_repository_impl.dart:
```diff
@@ -16,8 +16,11 @@ class DishesRepositoryImpl implements DishesRepository {
   @override
   Future<List<DishModel>> getFirstDishes(bool hasInternet) async {
-    List<DishModel> listResult = [];
+    final List<DishModel> listResult = [];
```
нейминг

AliaksandrHuzich commented on food_app/data/lib/repository/orders_repository_impl.dart:
```diff
@@ -0,0 +1,43 @@
+  final OrdersMapper _ordersMapper;
+
+  OrdersRepositoryImpl(
+    this._firestoreProvider,
```
именнованые и обязательные

AliaksandrHuzich commented on food_app/domain/lib/usecases/sign_up_with_email_and_password_usecase.dart:
```diff
@@ -1,16 +1,17 @@
   @override
   Future<void> execute(List<String> input) async {
-    return _authRepository.signUp(
+    return _authRepository.signUpWithEmailAndPassword(
```
модель

AliaksandrHuzich commented on food_app/features/auth_view/lib/src/bloc/auth_view/bloc.dart:
```diff
@@ -6,25 +6,28 @@ part 'event.dart';
+      required LogInUsecase logInUsecase,
+      required SignOutUsecase signOutUsecase,
+      required SignUpWithGoogleUsecase signUpWithGoogleUsecase,
+      required InitUserUsecase initUserUsecase})
```
,

AliaksandrHuzich commented on food_app/features/auth_view/lib/src/bloc/auth_view/bloc.dart:
```diff
@@ -34,27 +37,10 @@ class AuthViewBloc extends Bloc<AuthViewEvent, AuthViewState> {
     Emitter<AuthViewState> emit,
   ) async {
-    final bool? isLoggedIn;
+    final bool isLoggedIn;
```
в 1 строку

AliaksandrHuzich commented on food_app/features/cart_view/lib/src/bloc/cart_view/bloc.dart:
```diff
@@ -34,13 +35,61 @@ class CartViewBloc extends Bloc<CartViewEvent, CartViewState> {
+    final bool hasInternet =
+        await appLocator.get<InternetConnection>().hasInternetAccess;
+    try {
+      if (hasInternet) {
```
внутри репо

AliaksandrHuzich commented on food_app/features/cart_view/lib/src/widget/cart_list_view_item.dart:
```diff
@@ -18,73 +17,63 @@ class CartListViewItem extends StatelessWidget {
-                ],
+              CustomText(
+                text: '${itemModel.count} * ${itemModel.cost} = ${double.parse(
+                  (itemModel.count * itemModel.cost).toStringAsFixed(2),
```
вычисление в стейт

AliaksandrHuzich commented on food_app/features/cart_view/lib/src/widget/order_button.dart:
```diff
@@ -0,0 +1,56 @@
+              BlocProvider.of<CartViewBloc>(context).add(
+                ClearCartEvent(),
+              );
+              final bool hasInternet =
```
переписать

AliaksandrHuzich commented on food_app/features/dishes_view/lib/src/ui/dishes_view_screen.dart:
```diff
@@ -64,57 +72,69 @@ class _DishesViewScreenState extends State<DishesViewScreen> {
+                child: TypeListView(),
+              ),
+              Expanded(
+                flex: 12,
```
перебор

AliaksandrHuzich commented on food_app/features/dishes_view/lib/src/widget/type_list_tile.dart:
```diff
@@ -0,0 +1,35 @@
+          ),
+        ),
+      );
+    });
```
,

AliaksandrHuzich commented on food_app/features/dishes_view/lib/src/widget/type_list_view.dart:
```diff
@@ -0,0 +1,39 @@
+          ),
+        ),
+      );
+    });
```
,

AliaksandrHuzich commented on food_app/features/order_history_view/lib/src/ui/orders_view_screen.dart:
```diff
@@ -0,0 +1,97 @@
+                      );
+                    },
+                    separatorBuilder: (BuildContext context, int index) =>
+                        const SizedBox(
```
не стрелочная а ретурн

LonelyMidoriya committed:
[PT-6] Add order history

Add an "Order History" page
Implement filtering by categories
+
Some Fixes
Rewrite internet connection fuctionality
